language/bytecode-verifier/src/signature.rs

```diff
@@ -156,7 +156,75 @@ impl<'a> SignatureChecker<'a> {
                         type_parameters,
                     )
                 }
-                _ => Ok(()),
+                VecEmpty(idx) | VecLen(idx) | VecImmBorrow(idx) | VecMutBorrow(idx)
+                | VecPushBack(idx) | VecPopBack(idx) | VecDestroy(idx) | VecSwap(idx) => {
+                    let type_arguments = &self.resolver.signature_at(*idx).0;
+                    self.check_signature_tokens(type_arguments)?;
+                    self.check_generic_instance(
+                        type_arguments,
+                        &[AbilitySet::PRIMITIVES],
+                        type_parameters,
+                    )
+                }
+
+                // List out the other options explicitly so there's a compile error if a new
+                // bytecode gets added.
+                Pop
+                | Ret
+                | Branch(_)
+                | BrTrue(_)
+                | BrFalse(_)
+                | LdU8(_)
+                | LdU64(_)
+                | LdU128(_)
+                | LdConst(_)
+                | CastU8
+                | CastU64
+                | CastU128
+                | LdTrue
+                | LdFalse
+                | Call(_)
+                | Pack(_)
+                | Unpack(_)
+                | ReadRef
+                | WriteRef
+                | FreezeRef
+                | Add
+                | Sub
+                | Mul
+                | Mod
+                | Div
+                | BitOr
+                | BitAnd
+                | Xor
+                | Shl
+                | Shr
+                | Or
+                | And
+                | Not
+                | Eq
+                | Neq
+                | Lt
+                | Gt
+                | Le
+                | Ge
+                | CopyLoc(_)
+                | MoveLoc(_)
+                | StLoc(_)
+                | MutBorrowLoc(_)
+                | ImmBorrowLoc(_)
+                | MutBorrowField(_)
+                | MutBorrowFieldGeneric(_)
+                | ImmBorrowField(_)
+                | ImmBorrowFieldGeneric(_)
+                | MutBorrowGlobal(_)
+                | ImmBorrowGlobal(_)
+                | Exists(_)
+                | MoveTo(_)
+                | MoveToGeneric(_)
```

Comment:
Question: is there a reason why `MoveToGeneric`, `MutBorrowFieldGeneric`, `ImmBorrowFieldGeneric` were not checked originally?

Reply:
Looks like a bug to me. Good catch!

Reply:
I can put another PR to address this then. But let me know if you are already working on this.

Reply:
Go ahead and put up a PR! I'm not currently touching this part of the codebase so no need to worry about conflicts.